src/components/TodoItem.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+
+interface Props {
+  title: string,
+  id: number,
+  completed: boolean,
+  isLoading: boolean,
+  deleteTodos: (id: number) => void,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  id, completed, title, deleteTodos, isLoading = false,
+}) => {
+  const [loader, setLoader] = useState(false);
+
+  useEffect(() => {
+    setLoader(isLoading);
+  }, [isLoading]);
+
+  const handleDelete = (todoId: number) => {
+    deleteTodos(todoId);
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={`todo ${completed && 'completed'}`}
+      key={id}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          defaultChecked={completed}
+        />
+      </label>
+
+      <span data-cy="TodoTitle" className="todo__title">
+        {title}
+      </span>
+
+      {/* Remove button appears only on hover */}
+      <button
+        type="button"
+        className="todo__remove"
+        data-cy="TodoDelete"
+        onClick={() => handleDelete(id)}
+      >
+        ×
+      </button>
+
+      {/* overlay will cover the todo while it is being updated */}
+      <div
+        data-cy="TodoLoader"
+        className={`modal overlay ${
+          loader && 'is-active'
```

Comment:
better to use classnames package for such cases
<img width="450" alt="image" src="https://github.com/mate-academy/react_todo-app-add-and-delete/assets/70220667/848d357b-a26f-43f2-b059-3478b25cc8f2">